vignettes/base_R.Rmd

```diff
@@ -0,0 +1,235 @@
+
+# Direct translations
+Here are base R commands and equivalent `purrr` commands. However, note that the `purrr` alternative can differ from the base R version subtly. Please check function documentations for such subtleties.
+
+## `Map` functions
+Here `X` denotes a vector or a list and `FUN` denotes a function
+
+Output | Input | Base R | `purrr`
+-------|-------|--------|-----------
+List | 1 vector | `lapply` | `map`
+List | 2 vectors | `mapply`, `Map` | `map2`
+List | >2 vectors | `mapply`, `Map` | `pmap`
+Atomic vector of desired type | 1 vector | `vapply` | `map_lgl` (logical), `map_int` (integer), `map_dbl` (double), `map_chr` (character), `map_raw` (raw)
+Atomic vector of desired type | 2 vectors | `NA` | `map2_lgl` (logical), `map2_int` (integer), `map2_dbl` (double), `map2_chr` (character), `map2_raw` (raw)
+Atomic vector of desired type | >2 vectors | `NA` | `pmap_lgl` (logical), `pmap_int` (integer), `pmap_dbl` (double), `pmap_chr` (character), `pmap_raw` (raw)
+Side effect only | 1 vector | loops | `walk`
+Side effect only | 2 vectors | loops | `walk2`
+Side effect only | >2 vectors | loops | `pwalk`
+Data frame (`rbind` outputs) | 1 vector | `lapply` then `rbind` | `map_dfr`
+Data frame (`rbind` outputs) | 2 vectors | `mapply`/`Map` then `rbind` | `map2_dfr`
+Data frame (`rbind` outputs) | >2 vectors | `mapply`/`Map` then `rbind` | `pmap_dfr`
+Data frame (`cbind` outputs) | 1 vector | `lapply` then `cbind` | `map_dfc`
+Data frame (`cbind` outputs) | 2 vectors | `mapply`/`Map` then `cbind` | `map2_dfc`
+Data frame (`cbind` outputs) | >2 vectors | `mapply`/`Map` then `cbind` | `pmap_dfc`
+Any | Vector and its names | `l/s/vapply(X, function(x) FUN(x, names(x)))` or `mapply/Map(FUN, X, names(X))` | `imap`, `imap_*` (`lgl`, `dbl`, `dfr`, and etc. just like in `map`, `map2`, and `pmap`)
+Any | Selected elements of the vector | `l/s/vapply(X[index], FUN, ...)` | `map_if`, `map_at`
+List | Recursively apply to list within list | `rapply` | `map_depth`
+List | List only | `lapply` | `lmap`, `lmap_at`, `lmap_if`
+
+## Shorthands
+When an anonymous function is required in `*apply` or `map` functions, `purrr` offers shorthands to make the anonymous function more readable and easier to write. However, using shorthands is not recommended inside functions and packages. Here `l` denotes a list of arguments, and `f` denotes some expression involving arguments of the anonymous function.
+
+Number of inputs | base R | `purrr`
+:---:|--------|---------
+1 | `function(x) f(x)` | `~ f(.x)`
+2 | `function(x, y) f(x, y)` | `~ f(.x, .y)`
+>2 | `function(x, y, z, ...) f(x, y, z, ...)` or `function(l) f(l[[1]], l[[2]], l[[3]], ...)` or `function(l) do.call(f, args = l)` | `~ f(..1, ..2, ..3, ...)`
+
```

Comment:
Maybe mention the plucking shortcuts here too?

Reply:
OK